src/map/pc.c

```diff
@@ -8707,10 +8709,12 @@ bool pc_setcart(struct map_session_data *sd,int type) {
 			if( !sd->sc.data[SC_PUSH_CART] ) /* first time, so fill cart data */
 				clif_cartlist(sd);
 			clif_updatestatus(sd, SP_CARTINFO);
-			sc_start(&sd->bl,&sd->bl, SC_PUSH_CART, 100, type, 0);
+			sc_start(&sd->bl,&sd->bl, SC_PUSH_CART, 100, type, -1);
```

Comment:
It's not for set timer ID.
actually it's a consistency if follow `sc_start` script with **infinity timer** by using -1.